File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/GremlinWarrior_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class GremlinWarrior_Model extends MonsterInfo{
    public static Id = "GremlinWarrior"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(21, 25);
        else
            return Mathf.RandomRange(20, 24);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/angryGremlin/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/angryGremlin/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/angryGremlin/skeleton.png"));
    }

    get Name(): string {
        return "火大地精";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/GremlinWarrior_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class GremlinWarrior_Model : MonsterInfo
    {
        public static string Id = "GremlinWarrior";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(21, 25) : Random.Range(20, 24);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/angryGremlin/skeleton.atlas",
                    "Monsters/theBottom/angryGremlin/skeleton.json", "Monsters/theBottom/angryGremlin/skeleton.png");
            }
        }

        public override string Name => "火大地精";

        public GremlinWarrior_Model(Vector2 pos) : base(pos)
        {
        }
    }
}